### PR TITLE
Mangathemesia refactor

### PR DESCRIPTION
**Pull request by AwkwardPeak7**
- Remove Random UA; should be added on a per source basis.
- Use `:lib:i18n`
- add selectors for other languages
  - would like some comments on the approach
-  countViews in background

### CODE REVIEW

File: lib-multisrc/mangathemesia/src/eu/kanade/tachiyomi/multisrc/mangathemesia/MangaThemesia.kt [hunk 1]
```diff
@@ -210,23 +261,42 @@ abstract class MangaThemesia(
...
-        this == null -> SManga.UNKNOWN
-        listOf("ongoing", "publishing").any { this.contains(it, ignoreCase = true) } -> SManga.ONGOING
-        this.contains("hiatus", ignoreCase = true) -> SManga.ON_HIATUS
-        this.contains("completed", ignoreCase = true) -> SManga.COMPLETED
-        listOf("dropped", "cancelled").any { this.contains(it, ignoreCase = true) } -> SManga.CANCELLED
-        else -> SManga.UNKNOWN
+    open fun String?.parseStatus(): Int {
+        if (this == null) return SManga.UNKNOWN
+
+        return when (this) {
```
Comment (stevenyomi) [inline lib-multisrc/mangathemesia/src/eu/kanade/tachiyomi/multisrc/mangathemesia/MangaThemesia.kt]: Lowercase this?

File: lib-multisrc/mangathemesia/src/eu/kanade/tachiyomi/multisrc/mangathemesia/MangaThemesia.kt [hunk 2]
```diff
@@ -338,18 +404,25 @@ abstract class MangaThemesia(
...
-        runCatching { client.newCall(request).execute().close() }
+        try {
+            val request = countViewsRequest(document) ?: return
+            client.newCall(request).enqueue(countViewsCallback)
+        } catch (_: Exception) { }
+    }
+
+    private val countViewsCallback = object : Callback {
+        override fun onResponse(call: Call, response: Response) = response.close()
+        override fun onFailure(call: Call, e: IOException) = Unit
```
Comment (stevenyomi) [inline lib-multisrc/mangathemesia/src/eu/kanade/tachiyomi/multisrc/mangathemesia/MangaThemesia.kt]: Create the callback on site, it's cheap enough. And we don't need the `try-catch` block.

File: lib-multisrc/mangathemesia/src/eu/kanade/tachiyomi/multisrc/mangathemesia/MangaThemesia.kt (hunk 1 above)
Comment (AwkwardPeak7) [reply lib-multisrc/mangathemesia/src/eu/kanade/tachiyomi/multisrc/mangathemesia/MangaThemesia.kt]: idk if some of the lowercase versions of translations are correct

File: lib-multisrc/mangathemesia/src/eu/kanade/tachiyomi/multisrc/mangathemesia/MangaThemesia.kt (hunk 2 above)
Comment (stevenyomi) [reply lib-multisrc/mangathemesia/src/eu/kanade/tachiyomi/multisrc/mangathemesia/MangaThemesia.kt]: Could use `val callback = object : Callback {`, that's less indentation.

File: lib-multisrc/mangathemesia/src/eu/kanade/tachiyomi/multisrc/mangathemesia/MangaThemesia.kt (hunk 1 above)
Comment (stevenyomi) [reply lib-multisrc/mangathemesia/src/eu/kanade/tachiyomi/multisrc/mangathemesia/MangaThemesia.kt]: Could test in both AS and an Android device (print/log them). Maybe I could write a lint rule to help with this.

File: lib-multisrc/mangathemesia/src/eu/kanade/tachiyomi/multisrc/mangathemesia/MangaThemesia.kt [hunk 3]
```diff
@@ -165,23 +150,89 @@ abstract class MangaThemesia(
...
+
+    private val artistLabels = listOf(
+        "artist",
+        "Artiste",
+        "Artista",
+        "الرسام",
+        "الناشر",
+        "İllüstratör",
+        "Çizer",
+    )
```
Comment (stevenyomi) [inline lib-multisrc/mangathemesia/src/eu/kanade/tachiyomi/multisrc/mangathemesia/MangaThemesia.kt]: These lists of labels aren't used anywhere else right? Then inline them to save memory.

File: lib-multisrc/mangathemesia/src/eu/kanade/tachiyomi/multisrc/mangathemesia/MangaThemesia.kt [hunk 4]
```diff
@@ -210,23 +261,42 @@ abstract class MangaThemesia(
...
-        listOf("dropped", "cancelled").any { this.contains(it, ignoreCase = true) } -> SManga.CANCELLED
-        else -> SManga.UNKNOWN
+    open fun String?.parseStatus(): Int {
+        if (this == null) return SManga.UNKNOWN
+
+        return when (this.lowercase().trim()) {
+            "مستمرة", "en curso", "ongoing", "on going", "ativo", "en cours",
+            "en cours de publication", "Đang tiến hành", "em lançamento", "Онгоінг", "publishing",
+            "devam ediyor", "em andamento", "in corso", "güncel", "berjalan", "Продолжается", "updating", "lançando", "in arrivo", "emision",
+            "en emision", "مستمر", "curso", "en marcha", "publicandose", "publicando", "连载中", "devam etmekte", "連載中",
```
Comment (stevenyomi) [inline lib-multisrc/mangathemesia/src/eu/kanade/tachiyomi/multisrc/mangathemesia/MangaThemesia.kt]:
```suggestion
            "مستمرة", "en curso", "ongoing", "on going", "ativo", "en cours",
            "en cours de publication", "đang tiến hành", "em lançamento", "онгоінг", "publishing",
            "devam ediyor", "em andamento", "in corso", "güncel", "berjalan", "продолжается", "updating", "lançando", "in arrivo", "emision",
            "en emision", "مستمر", "curso", "en marcha", "publicandose", "publicando", "连载中", "devam etmekte", "連載中",
```

File: lib-multisrc/mangathemesia/src/eu/kanade/tachiyomi/multisrc/mangathemesia/MangaThemesia.kt [hunk 5]
```diff
@@ -210,23 +261,42 @@ abstract class MangaThemesia(
...
+            -> SManga.ONGOING
+
+            "completed", "completo", "complété", "fini", "achevé", "terminé", "tamamlandı", "Đã hoàn thành", "hoàn thành",
+            "مكتملة", "Завершено", "finished", "finalizado", "completata", "one-shot", "bitti", "tamat", "completado", "concluído", "完結",
+            "concluido", "已完结", "bitmiş",
```
Comment (stevenyomi) [inline lib-multisrc/mangathemesia/src/eu/kanade/tachiyomi/multisrc/mangathemesia/MangaThemesia.kt]:
```suggestion
            "completed", "completo", "complété", "fini", "achevé", "terminé", "tamamlandı", "đã hoàn thành", "hoàn thành",
            "مكتملة", "завершено", "finished", "finalizado", "completata", "one-shot", "bitti", "tamat", "completado", "concluído", "完結",
            "concluido", "已完结", "bitmiş",
```

File: lib-multisrc/mangathemesia/src/eu/kanade/tachiyomi/multisrc/mangathemesia/MangaThemesia.kt [hunk 6]
```diff
@@ -210,23 +261,42 @@ abstract class MangaThemesia(
...
+            -> SManga.COMPLETED
+
+            "canceled", "cancelled", "cancelado", "cancellato", "cancelados", "dropped", "discontinued", "abandonné",
+            -> SManga.CANCELLED
+
+            "hiatus", "on hold", "pausado", "en espera", "en pause", "en attente", "人気",
```
Comment (stevenyomi) [inline lib-multisrc/mangathemesia/src/eu/kanade/tachiyomi/multisrc/mangathemesia/MangaThemesia.kt]: Remove "人気" (which means popular). It's taken from https://manga-mate.org/ but that status is empty. Could remove it from MangaMate extension too.

**Pull request merged**